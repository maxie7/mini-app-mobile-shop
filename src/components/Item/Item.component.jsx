import React, {useEffect, useState} from 'react';
import PropTypes from 'prop-types';
import { ItemContainer } from "./Item.styles";
import { Paragraph, Frame } from "arwes";

const Item = ({ id, brand, model, price }) => {
  const [showFrame, setShowFrame] = useState(false);

  useEffect(() => {
    const timer = setTimeout(() => {
      setShowFrame(true);
    }, 600);
    return () => clearTimeout(timer);
  }, []);

  return (
    <Frame
      show={showFrame}
      animate={true}
      level={3}
      corners={4}
      layer='primary'
    >
      <ItemContainer key={ id }>
        <Paragraph>{ brand }</Paragraph>
        <p>{ model }</p>
        <p>{ price }</p>
      </ItemContainer>
    </Frame>
  )
}

Item.propTypes = {
  id: PropTypes.string,
  brand: PropTypes.string,
  model: PropTypes.string,
  price: PropTypes.string
};

export default Item;


