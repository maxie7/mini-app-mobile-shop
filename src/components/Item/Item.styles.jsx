import styled from 'styled-components';

export const ItemContainer = styled.div`
  background: #021114;
  display: flex;
  flex-direction: column;
  //background-color: rgba(149, 218, 218, 0.7);
  //border: 1px solid grey;
  //border-radius: 5px;
  //padding: 25px;
  cursor: pointer;
  -moz-osx-font-smoothing: grayscale;
  backface-visibility: hidden;
  padding: 20px 40px;
  transform: translateZ(0);
  transition: transform 0.25s ease-out;
  
  &:hover {
    transform: scale(1.1);
  }
`;
