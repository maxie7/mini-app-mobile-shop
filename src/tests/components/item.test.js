import React from "react";
import {shallow} from "enzyme";
import Item from "../../components/Item/Item.component";

it('should render Item section', () => {
  expect(shallow(<Item />)).toMatchSnapshot();
});