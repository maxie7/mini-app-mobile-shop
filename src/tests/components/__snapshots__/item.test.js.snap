// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Item section 1`] = `ShallowWrapper {}`;
